flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/SumAggFunction.scala

```diff
@@ -67,9 +80,9 @@ abstract class SumAggFunction[T: Numeric] extends AggregateFunction[T] {
     var i: Int = 0
     while (i < accumulators.size()) {
       val a = accumulators.get(i).asInstanceOf[SumAccumulator[T]]
-      if (a.f1) {
+      if (a.f1 > 0) {
```

Comment:
Do we need this check? If `a.f1` is `0`, `a.f0` should be `0` as well (if we only retract what was added before) and we can simply add sum and count.